### PR TITLE
release: add release script for version bumping and gem publishing

### PR DESCRIPTION
**Pull request by sacry-dyn**

### CODE REVIEW

**Comment by sacry-dyn** [comment]
@tobstarr could you review?

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+PHRASEAPP_RUBY_REPO=git@github.com:phrase/phraseapp-ruby.git
+PHRASEAPP_RUBY_TMP=$(mktemp -d /tmp/phraseapp-ruby-lib-XXXX)
+
+
+function cleanup {
+  rm -Rf $PHRASEAPP_RUBY_TMP
+}
+
+function write_version {
+cat <<EOF > lib/phraseapp-ruby/version.rb
```
Comment (tobstarr) [inline release.sh]: I prefer doing `cat > lib/phraseapp-ruby/version.rb <<EOF` for better syntax highlighting

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+# Usage
+# PUBLISH_GEM=true RELEASE_VERSION=1.2.7 ./release.sh
+
+PHRASEAPP_RUBY_REPO=git@github.com:phrase/phraseapp-ruby.git
+PHRASEAPP_RUBY_TMP=$(mktemp -d /tmp/phraseapp-ruby-lib-XXXX)
```
Comment (tobstarr) [inline release.sh]: I would probably try to check if that works `if [[ -z $PHRASE_RUBY_TMP]]; then echo "unable to create tmp dir"; exit 1; fi` as there could be an issue here (inodes, permissions, etc...)

File: release.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+EOF
+}
+
+function push_to_git {
+  current_branch=$(git symbolic-ref --short HEAD)
+  if [[ "$current_branch" != "$1" ]]; then
+    git checkout -b $1
+  fi
+
+  git add .
```
Comment (tobstarr) [inline release.sh:29]: what about other local changes? would it be not enough to just both generated files?

Comment (tobstarr) [inline release.sh:29]: ah, we do a fresh clone anyway?

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+trap "cleanup" EXIT
+
+git clone $PHRASEAPP_RUBY_REPO $PHRASEAPP_RUBY_TMP
+
+pushd $PHRASEAPP_RUBY_TMP > /dev/null
+
+# test -e fails
+rake test
+
+current_version=$(cat lib/phraseapp-ruby/version.rb | grep VERSION | cut -d "=" -f 2 | xargs | tr -d '\r')
```
Comment (tobstarr) [inline release.sh]: please also check for not blank (`test -z`)

File: release.sh (hunk 3 above)
Comment (sacry-dyn) [inline release.sh:29]: yes, every time the script is called

File: release.sh (hunk 2 above)
Comment (sacry-dyn) [inline release.sh]: good catch

File: release.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+# Usage
+# PUBLISH_GEM=true RELEASE_VERSION=1.2.7 ./release.sh
+
+PHRASE_RUBY_REPO=git@github.com:phrase/phraseapp-ruby.git
+PHRASE_RUBY_TMP=$(mktemp -d /tmp/phraseapp-ruby-lib-XXXX)
+
+trap "rm -Rf $PHRASE_RUBY_TMP" EXIT
+
+if [[ -z "$PHRASE_RUBY_TMP" ]]; then echo "unable to create tmp dir"; exit 1; fi
```
Comment (tobstarr) [inline release.sh:12]: we would need to return before the trap is set up. but: I realized the `-z` is not necessary as `set -e` would not continue anyway so you can just remove it again :)

Comment (sacry-dyn) [inline release.sh:12]: true. I will swap the trap with the -z better error output.

**Pull request merged**